### PR TITLE
feat(SpeechToText): Add developer docs for OCP\SpeechToText

### PR DESCRIPTION
**Pull request by marcelklehr**
server PR: https://github.com/nextcloud/server/pull/37674

### CODE REVIEW

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Comment by nickvergessen** [comment]
Can you add a screenshot of your rendered page so we can see how it looks?

**Comment by marcelklehr** [comment]
:grin: 

I was trying to avoid having to build it... I'll give it a try.

**Comment by marcelklehr** [comment]
ah, wasn't that hard.

image

File: developer_manual/digging_deeper/speech-to-text.rst [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ * ``transcribeFile(File $file)`` This method takes a ``OCP\Files\File`` Object which should point to a media file and will attempt to transcribe it **in the current process**. It will thus block until transcription has finished and will return the transcript as a string. Using this method is thus only recommend in CLI commands or Background Jobs when you are not limited by any HTTP request timeouts and execution time limits.
+ * ``scheduleFileTranscription(File $file, ?string $userId, string $appId)`` This method schedules a transcription of the passed media file **in a background job** and will thus not block.
+
+Listening to the transcription events
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since ``scheduleFileTranscription`` does not block, you will need to listen to the following events in your app to obtain the transcript or be notified of any failure.
+
+ * ``OCP\SpeechToText\Events\TranscriptionSuccessfulEvent`` This event class offers the ``getTranscript()`` method which returns the transcript as a string
+ * ``OCP\SpeechToText\Events\TranscriptionFailedEvent`` This event class offers the ``getErrorMessage()`` method which returns the error message as a string
```
Comment (nickvergessen) [inline developer_manual/digging_deeper/speech-to-text.rst:24]: Is this ready for display (so in user language) or neutral/english and the consumers need to overwrite it with something generic?

File: developer_manual/digging_deeper/speech-to-text.rst [hunk 2]
```diff
@@ -0,0 +1,132 @@
+.. _speech-to-text:
+
+==============
+Speech-To-Text
+==============
+
+Nextcloud 27 offers a new **Speech-To-Text** API. The overall idea is that there is a central OCP API that apps can use to request transcriptions of audio or video files. To be technology agnostic any app can provide this Speech-To-Text functionality by registering a Speech-To-Text provider.
```
Comment (ChristophWurst) [inline developer_manual/digging_deeper/speech-to-text.rst:7]:
```suggestion
==============

.. versionadded:: 27

Nextcloud offers a **Speech-To-Text** API. The overall idea is that there is a central OCP API that apps can use to request transcriptions of audio or video files. To be technology agnostic any app can provide this Speech-To-Text functionality by registering a Speech-To-Text provider.
```

https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded

File: developer_manual/digging_deeper/speech-to-text.rst [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+Listening to the transcription events
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since ``scheduleFileTranscription`` does not block, you will need to listen to the following events in your app to obtain the transcript or be notified of any failure.
+
+ * ``OCP\SpeechToText\Events\TranscriptionSuccessfulEvent`` This event class offers the ``getTranscript()`` method which returns the transcript as a string
+ * ``OCP\SpeechToText\Events\TranscriptionFailedEvent`` This event class offers the ``getErrorMessage()`` method which returns the error message as a string
+
+Both classes provide the ``$appId` and ``$userId`` params that you initially passed to ``scheduleFileTranscription`` via ``getAppId()`` and ``getUserId()`` as well as ``getFileId()`` and ``getFile()`` to access the media file that was transcribed.
```
Comment (nickvergessen) [inline developer_manual/digging_deeper/speech-to-text.rst:26]:
```suggestion
Both classes provide the ``$appId`` and ``$userId`` params that you initially passed to ``scheduleFileTranscription`` via ``getAppId()`` and ``getUserId()`` as well as ``getFileId()`` and ``getFile()`` to access the media file that was transcribed.
```

File: developer_manual/digging_deeper/speech-to-text.rst (hunk 1 above)
Comment (marcelklehr) [reply developer_manual/digging_deeper/speech-to-text.rst:24]: It's only for debugging purposes, I'd say. I'll add that

File: developer_manual/digging_deeper/speech-to-text.rst (hunk 2 above)
Comment (marcelklehr) [reply developer_manual/digging_deeper/speech-to-text.rst:7]: huh, nice :blue_heart:

File: developer_manual/digging_deeper/speech-to-text.rst [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    use OCA\MyApp\SpeechToText\Provider;
+    use OCP\AppFramework\App;
+    use OCP\AppFramework\Bootstrap\IBootContext;
+    use OCP\AppFramework\Bootstrap\IBootstrap;
+    use OCP\AppFramework\Bootstrap\IRegistrationContext;
+
+    class Application extends App implements IBootstrap {
+
+        public function register(IRegistrationContext $context): void {
+            $context->registerSpeechToTextProvider(Provider::class);
```
Comment (ChristophWurst) [inline developer_manual/digging_deeper/speech-to-text.rst:135]: highlight this line with `:emphasize-lines:` https://docs.nextcloud.com/server/latest/developer_manual/basics/middlewares.html#global-middlewares

**Comment by marcelklehr** [comment]
<details>
<summary>Screenshot</summary>

image


</details>

**Comment by marcelklehr** [comment]
<details>
<summary>Screenshot</summary>

image


</details>

File: developer_manual/digging_deeper/speech-to-text.rst (hunk 4 above)
Comment (nickvergessen) [reply developer_manual/digging_deeper/speech-to-text.rst:135]: TIL, nice!

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**